### PR TITLE
[Merged by Bors] - chore: Move scalar compatibility instance for `ℤ` and `ℕ` on rings to their own files

### PR DESCRIPTION
**Pull request by YaelDillies**
Part of #9411.

Also corrects some instance names in the docstrings.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/GroupTheory/GroupAction/Ring.lean [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Those instances are in their own file only because they require much less imports than any existing
+file they could go to. This is unfortunate and should be fixed by reorganising files.
+-/
+
+open scoped Int
+
+variable {α : Type*}
+
+/-- Note that `AddCommMonoid.nat_smulCommClass` requires stronger assumptions on `α`. -/
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/GroupAction/Ring.lean:24]: This instance doesn't exist with this name

Comment (eric-wieser) [reply Mathlib/GroupTheory/GroupAction/Ring.lean:24]:
```suggestion
/-- Note that `AddMonoid.nat_smulCommClass` requires stronger assumptions on `α`. -/
```

File: Mathlib/GroupTheory/GroupAction/Ring.lean [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/-- Note that `AddCommMonoid.nat_isScalarTower` requires stronger assumptions on `α`. -/
+instance NonUnitalNonAssocSemiring.nat_isScalarTower [NonUnitalNonAssocSemiring α] :
+    IsScalarTower ℕ α α where
+  smul_assoc n x y := by
+    induction' n with n ih
+    · simp [zero_nsmul]
+    · simp_rw [succ_nsmul, ← ih, smul_eq_mul, add_mul]
+#align non_unital_non_assoc_semiring.nat_is_scalar_tower NonUnitalNonAssocSemiring.nat_isScalarTower
+
+/-- Note that `AddCommGroup.int_smulCommClass` requires stronger assumptions on `α`. -/
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/GroupAction/Ring.lean:42]:
```suggestion
/-- Note that `AddGroup.int_smulCommClass` requires stronger assumptions on `α`. -/
```

I think the `isScalarTower` names are already correct.

**Review by eric-wieser — APPROVED** [review]

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**